### PR TITLE
DAOS-11394 test: Save dmg log for daos_test suite

### PR DESCRIPTION
**Pull request by kjacque**
Save dmg logs for all daos_test cases.

Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [x] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [x] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [x] You are the appropriate gatekeeper to be landing the patch.
* [x] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [x] Githooks were used. If not, request that user install them and check copyright dates.
* [x] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [x] All builds have passed.  Check non-required builds for any new compiler warnings.
* [x] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [x] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [x] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Review by daltonbohning — CHANGES_REQUESTED** [review]
This test also uses the shared code, and unfortunately is not tagged with `pr` or `daos_test`
```
Test-tag: CsumErrorLog
```
> Test tag daos_test is covered by the standard PR tests. I saw the dmg results collected for tag daos_test on its own, so I am re-running with the full PR suite now for landing purposes

That's not entirely true. There are tests using this shared code, and tagged with `daos_test`, but NOT tagged with `pr`.

To run `pr`, `daos_test`, and `CsumErrorLog` you can do:
```
Features: daos_test CsumErrorLog
```

**Comment by kjacque** [comment]
@daltonbohning Good catches, thanks.

File: src/tests/ftest/util/daos_core_base.py [hunk 1]
```diff
@@ -148,3 +150,10 @@ def run_subtest(self):
                         rank, ["Stopped", "Excluded"])
 
         cmocka_utils.run_cmocka_test(self, job)
+
+        try:
+            tmp_log_path = "/tmp/suite_dmg.log"
+            log_path = os.path.join(self.outputdir, f"{self.subtest_name}_dmg.log")
+            shutil.move(tmp_log_path, log_path)
+        except IOError as error:
+            print("unable to move dmg log:", error)
```
[the remark this answers is not in the document]
Comment (kjacque) [reply src/tests/ftest/util/daos_core_base.py]: Good catch. I added a specific case for the FileNotFoundError as well. There seem to be a few tests where dmg isn't run by daos_test, so there isn't a log file.

**Review by daltonbohning — APPROVED** [review]
LGTM - thanks!

**Review by phender — APPROVED** [review]
As an improvement, it would be good to able to redefine the `/tmp/suite_dmg.log` to `os.path.join(self.outputdir, f"{self.subtest_name}_dmg.log")` prior to running the `daos_test` command to avoid the additional file moving code.  This would avoid potential problems with tests timeouts that prevent the file moving code from being executed or situations where the tests in run in a different environment where `/tmp` might not be an ideal location.

**Pull request merged**